contracts/utils/convex/ConvexStakingWrapper.sol

```diff
@@ -0,0 +1,452 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+pragma experimental ABIEncoderV2;
+
+import '@yield-protocol/utils-v2/contracts/token/IERC20.sol';
+import '@yield-protocol/utils-v2/contracts/token/ERC20.sol';
+import '@yield-protocol/utils-v2/contracts/token/TransferHelper.sol';
+import './interfaces/IRewardStaking.sol';
+import './interfaces/IConvexDeposits.sol';
+import "./interfaces/ICvx.sol";
+
+library CvxMining{
+    ICvx public constant cvx = ICvx(0x4e3FBD56CD56c3e72c1403e103b45Db9da5B9D2B);
+
+    function ConvertCrvToCvx(uint256 _amount) internal view returns(uint256){
+        uint256 supply = cvx.totalSupply();
+        uint256 reductionPerCliff = cvx.reductionPerCliff();
+        uint256 totalCliffs = cvx.totalCliffs();
+        uint256 maxSupply = cvx.maxSupply();
+
+        uint256 cliff = supply / reductionPerCliff;
+        //mint if below total cliffs
+        if(cliff < totalCliffs){
+            //for reduction% take inverse of current cliff
+            uint256 reduction = totalCliffs - cliff;
+            //reduce
+            _amount = _amount * reduction / totalCliffs;
+
+            //supply cap check
+            uint256 amtTillMax = maxSupply - supply;
+            if(_amount > amtTillMax){
+                _amount = amtTillMax;
+            }
+
+            //mint
+            return _amount;
+        }
+        return 0;
+    }
+}
+
+contract ConvexStakingWrapper is ERC20 {
+    using TransferHelper for IERC20;
+
+    struct EarnedData {
+        address token;
+        uint256 amount;
+    }
+
+    struct RewardType {
+        address reward_token;
+        address reward_pool;
+        uint128 reward_integral;
+        uint128 reward_remaining;
+        mapping(address => uint256) reward_integral_for;
+        mapping(address => uint256) claimable_reward;
+    }
+
+    uint256 public cvx_reward_integral;
+    uint256 public cvx_reward_remaining;
+    mapping(address => uint256) public cvx_reward_integral_for;
+    mapping(address => uint256) public cvx_claimable_reward;
+
+    //constants/immutables
+    address public constant convexBooster = address(0xF403C135812408BFbE8713b5A23a04b3D48AAE31);
+    address public constant crv = address(0xD533a949740bb3306d119CC777fa900bA034cd52);
+    address public constant cvx = address(0x4e3FBD56CD56c3e72c1403e103b45Db9da5B9D2B);
+    address public curveToken;
+    address public convexToken;
+    address public convexPool;
+    uint256 public convexPoolId;
+    address public collateralVault;
+
+    //rewards
+    RewardType[] public rewards;
+
+    //management
+    bool public isShutdown;
+    bool public isInit;
+    address public owner;
+
+    string internal _tokenname;
+    string internal _tokensymbol;
+
+    uint256 private constant _NOT_ENTERED = 1;
+    uint256 private constant _ENTERED = 2;
+
+    uint256 private _status;
```

Comment:
minor: _status could be `bool` and moved up after line 79 - I believe this will lead to better storage packing (fewer slots used)